nemo/deploy/deploy_pytriton.py

```diff
@@ -0,0 +1,182 @@
+        """
+        A nemo checkpoint or model is expected for serving on Triton Inference Server.
+
+        Args:
+            triton_model_name (str): Name for the service
+            triton_model_version(int): Version for the service
+            checkpoint_path (str): path of the nemo file
+            model (ITritonDeployable): A model that implements the ITritonDeployable from nemo.deploy import ITritonDeployable
+            max_batch_size (int): max batch size
+            port (int) : port for the Triton server
+            address (str): http address for Triton server to bind.
+        """
+
+        super().__init__(
+            triton_model_name=triton_model_name,
+            triton_model_version=triton_model_version,
+            checkpoint_path=checkpoint_path,
+            model=model,
+            max_batch_size=max_batch_size,
+            port=port,
+            address=address,
+            allow_grpc=allow_grpc,
+            allow_http=allow_http,
+            streaming=streaming,
+            pytriton_log_verbose=pytriton_log_verbose,
+        )
+
+    def deploy(self):
+
+        """
+        Deploys any models to Triton Inference Server.
+        """
+
+        self._init_nemo_model()
+
+        try:
+            if self.streaming:
+                # TODO: can't set allow_http=True due to a bug in pytriton, will fix in latest pytriton
+                triton_config = TritonConfig(
+                    log_verbose=self.pytriton_log_verbose,
+                    allow_grpc=self.allow_grpc,
+                    allow_http=self.allow_http,
+                    grpc_address=self.address,
+                )
+                self.triton = Triton(config=triton_config)
+                self.triton.bind(
+                    model_name=self.triton_model_name,
+                    model_version=self.triton_model_version,
+                    infer_func=self.model.triton_infer_fn_streaming,
+                    inputs=self.model.get_triton_input,
+                    outputs=self.model.get_triton_output,
+                    config=ModelConfig(decoupled=True),
+                )
+            else:
+                triton_config = TritonConfig(
+                    http_address=self.address,
+                    http_port=self.port,
+                    allow_grpc=self.allow_grpc,
+                    allow_http=self.allow_http,
+                )
+                self.triton = Triton(config=triton_config)
+                self.triton.bind(
+                    model_name=self.triton_model_name,
+                    model_version=self.triton_model_version,
+                    infer_func=self.model.triton_infer_fn,
+                    inputs=self.model.get_triton_input,
+                    outputs=self.model.get_triton_output,
+                    config=ModelConfig(max_batch_size=self.max_batch_size),
+                )
+        except Exception as e:
+            self.triton = None
+            print(e)
+
+    def serve(self):
```

Comment:
Whats the logical separation of deploy vs serve?

Reply:
That's mostly coming from the PyTriton. deploy function generates the PyTriton config files automatically. When we call serve (blocking) or run (non-blocking), it starts serving the model.

Will add docstrings.